lib/protobuf/wire.ex

```diff
@@ -71,7 +71,16 @@ defmodule Protobuf.Wire do
   @spec encode(proto_type(), proto_value()) :: iodata()
   def encode(type, value)
 
-  def encode(type, binary) when is_binary(binary) and type in [:string, :bytes],
+  def encode(:string, binary) when is_binary(binary) do
+    unless String.valid?(binary) do
+      raise Protobuf.EncodeError,
+        message: "#{inspect(binary)} is not valid UTF-8 for type string"
```

Comment:
Nitpick, but this might potentially be hard to read with bigger binaries.

```suggestion
      raise Protobuf.EncodeError,
        message: "invalid UTF-8 data for type string: #{inspect(binary)}"
```